src/geom.cpp

```diff
@@ -53,11 +53,14 @@ void simplify(GeometryType const &input, GeometryType &output, double max_distan
 			boost::geometry::index::query(outer_rtree, boost::geometry::index::nearest(line, nearest_query_size), std::back_inserter(nearest));
 
 			double min_distance = std::numeric_limits<double>::max();
-			for(auto const &i: nearest) 
-				min_distance = std::min(min_distance, boost::geometry::distance(line, i));
+			for(auto const &i: nearest) {
+				double dist = boost::geometry::distance(line, i);
+				if (dist>0.0) min_distance = std::min(min_distance, dist);
+			}
+		}
 
             std::size_t query_expected = ((start == 0 || end == input.size() - 1) ? 2 : 4);
-            if(result.size() == query_expected && min_distance < max_distance) {
+            if(result.size() == query_expected && min_distance > close_tolerance) {
```

Comment:
Nearest tolerance instead of close tolerance?

Reply:
Yep, that would work as a name!